src/mrinufft/operators/base.py

```diff
@@ -191,6 +152,35 @@ def with_off_resonnance_correction(self, B, C, indices):
 
         return MRIFourierCorrected(self, B, C, indices)
 
+    def compute_density(self, method=None, **kwargs):
+        """Compute the density compensation weights and set it.
+
+        Parameters
+        ----------
+        method: str or callable or array
+            The method to use to compute the density compensation.
+            If a string, the method should be registered in the density registry.
+            If a callable, it should take the samples and the shape as input.
+            If an array, it should be of shape (Nsamples,) and will be used as is.
+        """
+        if isinstance(method, np.ndarray):
+            self.density = method
+            return None
+        if not method:
+            self.density = None
+            return None
+
+        if method == "pipe" and "backend" not in kwargs:
+            method = self.pipe  # will raise error if not defined.
+        if method == "pipe" and "backend" in kwargs:
+            warnings.warn(f"Using pipe with {kwargs['backend']}")
```

Comment:
Why warn about it? It is more of info/log right?

Reply:
Indeed, but we are not (yet) using the logging module to provide different level of logging.